File: src/Model/Expression/ConstantExpression.java
package Model.Expression;

import DataTypes.MyIDictionary;
import Exception.ExpressionEvaluationException;

/**
 * Created by vladc on 12.10.2016.
 */
public class ConstantExpression extends Expression {
    private int number;

    public ConstantExpression(int number) {
        this.number = number;
    }

    public int evaluate(MyIDictionary<String, Integer> symTab) throws ExpressionEvaluationException {
        return number;
    }

    @Override
    public String toString() {
        return "" + number;
    }
}

File: Toy-Language-Interpreter_C#/Toy-Language-Interpreter/Model/Expression/ConstantExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.DataTypes;

namespace Toy_Language_Interpreter.Model.Expression
{
    class ConstantExpression : AExpression
    {
        private int number;

        public ConstantExpression(int number)
        {
            this.number = number;
        }

        public override int Evaluate(MyIDictionary<string, int> symTab)
        {
            return number;
        }

        public override string ToString()
        {
            return "" + number;
        }

    }
}
